otbox.py

```diff
@@ -110,7 +142,7 @@ def _on_mqtt_connect(self, client, userdata, flags, rc):
 
         # subscribe to box commands
         client.subscribe('{0}/#'.format(self.mqtttopic_box_cmd_prefix))
-        client.subscribe('opentestbed/deviceType/box/deviceId/all/cmd/#')
+        client.subscribe('{0}/deviceType/box/deviceId/all/cmd/#'.format(self.args.testbed))
```

Comment:
Do you ensure somehow that otbox running on IoT-lab is not subscribed to e.g. `picturetoscreen` topic and similar, as it does not support that feature. My current understanding is that if `picturetoscreen` message arrives, it will invoke the handler even though otbox is running on IoT-lab, which will cause a crash due to lines 52-55. Right?

Reply:
The subscription to all the cmd topics is done via a wildcard, so yes it will subscribe to commands such as `picturetoscreen` even when the selected testbed is IoT-LAB. That's why I initially used an `if self.args.testbed == "opentestbed":` selection within the handlers of these commands to ensure that the crash doesn't occur if such a command is sent by accident (or a malicious intent). We argued that a command like this would not be sent by experiment controller when the first topic level is `iotlab`, but still, it's not a bad idea to cover all the exceptions. What I suggest is that we either unsubscribe from certain command topics immediately upon subscription if the selected testbed is not OpenTestbed, put back the removed “if” selections, or surround the handlers’ code with try-except. I prefer the first solution, as it seems the most elegant of the three.

Reply:
I remember the discussion but we also spoke about selective topic subscriptions to replace conditions in each handler. Why would we subscribe to a topic just to later unsubscribe? Could you not separate the features/topics to testbed-agnostic and testbed-specific and modify this code not to subscribe with a wildcard but rather something that takes into account a given testbed features?

Reply:
Why not? That way we introduce least amount of modifications to the code. It's the least "invasive" method, so to say, and the most elegant one in terms of number of lines. And, as far as I can see, here it's only a matter of OpenTestbed vs other testbeds, since it's only OpenTestbed that has these specific features. We would just check if the selected testbed is `opentestbed`. If not, cancel subscription for the OpenTestbed specific features.

Reply:
We apparently define "elegant" differently :-). What you propose will work for now so you should go for it. If we start having more similar "testbed-specific features" down the road, I guess we can reopen this discussion. What is important for me now is that there are no active subscriptions to topics whose handlers can cause crash in a particular environment.